packages/fetchai/connections/p2p_libp2p/connection.py

```diff
@@ -705,17 +702,17 @@ async def _receive_from_node(self) -> None:
         """
         while True:
             data = await self.node.read()
-            if data is None:
-                break
             assert self._in_queue is not None, "Input queue not initialized."
             self._in_queue.put_nowait(data)
+            if data is None:
```

Comment:
Not sure why this was moved here; data is already used in line 706

Reply:
Now you might be putting `None` on the queue

Reply:
yes, exactly. 
The `_receive_from_node` task will exit and the multiplexer will receive None on its scheduled `receive`